### PR TITLE
GHA/windows: add MSVC vcpkg MSH3 job

### PR DESCRIPTION
**Pull request by talregev**
Add msvc msh3 job
Fix compilation warning / error on test: Unreachable code.
Test 1014 failing. Should fix.

Http2 don't compile with msh3. Is it by design? 

Compilation error:
```
D:\a\curl\curl\lib\vquic\curl_msh3.c(251,13): error C2084: function 'void drain_stream(Curl_cfilter *,Curl_easy *,h2_stream_ctx *)' already has a body [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
      D:\a\curl\curl\lib\http2.c(352,13):
      see previous definition of 'drain_stream'
  
D:\a\curl\curl\lib\vquic\curl_msh3.c(599,7): error C2198: 'void drain_stream(Curl_cfilter *,Curl_easy *,h2_stream_ctx *)': too few arguments for call [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
  
D:\a\curl\curl\lib\vquic\curl_msh3.c(725,7): error C2198: 'void drain_stream(Curl_cfilter *,Curl_easy *,h2_stream_ctx *)': too few arguments for call [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
  
D:\a\curl\curl\lib\vquic\curl_msh3.c(7[29](https://github.com/curl/curl/actions/runs/10873342646/job/30169351365?pr=14922#step:10:29),7): error C2198: 'void drain_stream(Curl_cfilter *,Curl_easy *,h2_stream_ctx *)': too few arguments for call [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
  
D:\a\curl\curl\lib\vquic\curl_msh3.c(752,7): error C2198: 'void drain_stream(Curl_cfilter *,Curl_easy *,h2_stream_ctx *)': too few arguments for call [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
  
D:\a\curl\curl\lib\vquic\curl_msh3.c(764,5): error C2198: 'void drain_stream(Curl_cfilter *,Curl_easy *,h2_stream_ctx *)': too few arguments for call [D:\a\curl\curl\bld\lib\libcurl_object.vcxproj]
  (compiling source file 'CMakeFiles/libcurl_object.dir/Unity/unity_0_c.c')
```

Depends:
- [x] https://github.com/curl/curl/pull/14971
- [x] https://github.com/curl/curl/pull/14932
- [x] https://github.com/curl/curl/pull/14927

Refs: #14927 #14932 #14971

### CODE REVIEW

File: tests/unit/unit1660.c [hunk 1]
```diff
@@ -41,7 +41,7 @@ unit_stop(void)
 #if defined(CURL_DISABLE_HTTP) || defined(CURL_DISABLE_HSTS)
 UNITTEST_START
 {
-  return CURLE_OK; /* nothing to do when HTTP or HSTS are disabled */
+  /* return CURLE_OK;  nothing to do when HTTP or HSTS are disabled */
```
Comment (vszakats) [inline tests/unit/unit1660.c:44]: A fix has been pendig for this here:
https://github.com/curl/curl/pull/14772/files#diff-aa6618147822be7a9ddb938893eaa6c9e8e7216a9e875147eb1bd1e5708143cd

**Comment by vszakats** [comment]
> Add msvc msh3 job Fix compilation warning / error on test: Unreachable code. Test 1014 failing. Should fix.
> 
> Http2 don't compile with msh3. Is it intention?

The vcpkg package is not installed, isn't that the reason?

**Comment by talregev** [comment]
> > Add msvc msh3 job Fix compilation warning / error on test: Unreachable code. Test 1014 failing. Should fix.
> > Http2 don't compile with msh3. Is it intention?
> 
> The nghttp2 package is not installed, isn't that the reason?

No. I intentionally not install it. When I will add it, it will create a compilation error.
Do you want to see the error?
I can add it. I don't remember it.

File: tests/unit/unit1660.c (hunk 1 above)
Comment (talregev) [reply tests/unit/unit1660.c:44]: Let me know when it merge, and I will rebase.

**Comment by vszakats** [comment]
> > > Add msvc msh3 job Fix compilation warning / error on test: Unreachable code. Test 1014 failing. Should fix.
> > > Http2 don't compile with msh3. Is it intention?
> > 
> > 
> > The nghttp2 package is not installed, isn't that the reason?
> 
> No. I intentionally not install it. When I will add it, it will create a compilation error. Do you want to see the error? I can add it. I don't remember it.

Yep, I don't know much about MSH3 (tried building it once on non-Windows and failed), but on a quick glance it looks like a library that is a QUIC/H3 backend but not a general TLS backend, for H2 or H1 to use.

So it seems by design, but someone please correct me if wrong.

**Comment by talregev** [comment]
> tried building it once on non-Windows and failed

I can build it with vcpkg on non windows platform as well (linux, osx, other ...)

**Comment by vszakats** [comment]
Ah OK, thanks, it's always useful to include the actual failures.

...these ones are known, and pending fixes in this PR: #14815

**Comment by talregev** [comment]
> Ah OK, thanks, it's always useful to include the actual failures.
> 
> ...these ones are known, and pending fixes in this PR: #14815

Thanks! I added as depend PR.

**Comment by bagder** [comment]
> looks like a library that is a QUIC/H3 backend but not a general TLS backend, for H2 or H1 to use.

msh3 is a HTTP/3 library (that uses msquic). It does no other HTTP version.

**Comment by talregev** [comment]
@vszakats Can you fix test 1014 as well?

**Comment by vszakats** [comment]
The two required PRs are merged now. #14772 and #14815 are not required for this PR anymore.

**Comment by vszakats** [comment]
Dang, I was wrong, the unreachable code issue in unit1660 still needs one of the open PRs.

**Comment by vszakats** [comment]
Split that patch off to #14971.

**Comment by talregev** [comment]
> Dang, I was wrong, the unreachable code issue in unit1660 still needs one of the open PRs.
> 
> Yep, that's in https://github.com/curl/curl/pull/14772/files#diff-aa6618147822be7a9ddb938893eaa6c9e8e7216a9e875147eb1bd1e5708143cd. I'm gathering stamina for that one, there is a difficult-looking torture test issue in it, that needs to be tracked down first.

Can you make  a PR only fix the test?

**Comment by vszakats** [comment]
> > Dang, I was wrong, the unreachable code issue in unit1660 still needs one of the open PRs.
> > Yep, that's in https://github.com/curl/curl/pull/14772/files#diff-aa6618147822be7a9ddb938893eaa6c9e8e7216a9e875147eb1bd1e5708143cd. I'm gathering stamina for that one, there is a difficult-looking torture test issue in it, that needs to be tracked down first.
> 
> Can you make a PR only fix the test?

Not until I figure out what the problem is.

**Comment by vszakats** [comment]
unit1660 fix is merged.

**Comment by talregev** [comment]
> > > Dang, I was wrong, the unreachable code issue in unit1660 still needs one of the open PRs.
> > > Yep, that's in https://github.com/curl/curl/pull/14772/files#diff-aa6618147822be7a9ddb938893eaa6c9e8e7216a9e875147eb1bd1e5708143cd. I'm gathering stamina for that one, there is a difficult-looking torture test issue in it, that needs to be tracked down first.
> > 
> > 
> > Can you make a PR only fix the test?
> 
> Not until I figure out what the problem is.

I don't understand what problem are you referring to? 

I was refer to unreachable code compilation error, and you already create a seperate PR for it, as I requested.

**Comment by talregev** [comment]
All dependencies solved. 
@vszakats can you review my PR?

**Comment by vszakats** [comment]
> > > > Dang, I was wrong, the unreachable code issue in unit1660 still needs one of the open PRs.
> 
> > > > Yep, that's in https://github.com/curl/curl/pull/14772/files#diff-aa6618147822be7a9ddb938893eaa6c9e8e7216a9e875147eb1bd1e5708143cd. I'm gathering stamina for that one, there is a difficult-looking torture test issue in it, that needs to be tracked down first.
> 
> > > 
> 
> > > 
> 
> > > Can you make a PR only fix the test?
> 
> > 
> 
> > Not until I figure out what the problem is.
> 
> 
> 
> I don't understand what problem are you referring to? 
> 
> 
> 
> I was refer to unreachable code compilation error, and you already create a seperate PR for it, as I requested.  

Okay, by the time you "requested" this, the PR was already made, that why I didn't understand your question

**Comment by talregev** [comment]
ok. I didn't notice. Sorry for the confusion.

**Review by vszakats — APPROVED** [review]

**Comment by talregev** [comment]
@vszakats Are you thinking also to create msh3 in vcpkg in other os? (linux, osx)

**Comment by vszakats** [comment]
I tend to think this MSH3 job is enough to cover it for curl.

Perhaps a mingw-w64 vcpg build-only job, would be a combination interesting, if doable without too much hassle and runs fast in CI.

But my plate is totally full, so not so keen on breaking new grounds now.

**Comment by vszakats** [comment]
Thanks!

**Pull request closed without merging**